File: src/main/java/kz/chesschicken/astolfomod/item/ArmorAstolfo.java
package kz.chesschicken.astolfomod.item;

import net.minecraft.item.ItemInstance;
import net.modificationstation.stationapi.api.client.item.ArmorTextureProvider;
import net.modificationstation.stationapi.api.registry.Identifier;
import net.modificationstation.stationapi.api.template.item.armour.TemplateArmour;

public class ArmorAstolfo extends TemplateArmour implements ArmorTextureProvider {
    private final int type;
    private static final int[] BASE_DURABILITY = new int[] { 11, 16, 15, 13 };

    public ArmorAstolfo(Identifier id, int slot, int armorType) {
        super(id, 3, 3, slot);
        this.setDurability(BASE_DURABILITY[slot] * 3 << 3);
        this.maxStackSize = 1;
        this.type = armorType;
    }

    @Override
    public String getChestplateModelTexture(ItemInstance itemInstance) {
        switch (type)
        {
            case 0:
                return "/assets/astolfomod/textures/armor/astolfo_armored_2.png";
            case 1:
                return "/assets/astolfomod/textures/armor/astolfo_lightweight_2.png";
            default:
                return null;
        }
    }

    @Override
    public String getOtherModelTexture(ItemInstance itemInstance) {
        switch (type)
        {
            case 0:
                return "/assets/astolfomod/textures/armor/astolfo_armored_1.png";
            case 1:
                return "/assets/astolfomod/textures/armor/astolfo_lightweight_1.png";
            default:
                return null;
        }
    }
}

File: src/main/scala/kz/chesschicken/astolfomod/item/ArmorAstolfo.scala
package kz.chesschicken.astolfomod.item

import kz.chesschicken.astolfomod.ModListener
import kz.chesschicken.astolfomod.item.ArmorAstolfo.BASE_DURABILITY
import net.minecraft.item.armour.Armour
import net.modificationstation.stationapi.api.client.item.ArmourTextureProvider
import net.modificationstation.stationapi.api.registry.Identifier
import net.modificationstation.stationapi.api.template.item.armour.TemplateArmour
import net.minecraft.item.ItemInstance

object ArmorAstolfo {
  private val BASE_DURABILITY = Array[Int](11, 16, 15, 13)
}

class ArmorAstolfo(identifier: Identifier, slot: Int, armorType: Int) extends TemplateArmour(identifier, 3, 3, slot) with ArmourTextureProvider {
  this.setDurability(BASE_DURABILITY(slot) * (4 + (armorType * -1)) << 3)
  this.maxStackSize = 1

  override def getTexture(armour: Armour): Identifier = {
    armorType match {
      case 0 => Identifier.of(ModListener.publicModID, "astolfo_armored")
      case 1 => Identifier.of(ModListener.publicModID, "astolfo_lightweight")
    }
  }
}
